query/math.go

```diff
@@ -22,7 +22,7 @@ func (MathTypeMapper) MapType(measurement *influxql.Measurement, field string) i
 
 func (MathTypeMapper) CallType(name string, args []influxql.DataType) (influxql.DataType, error) {
 	switch name {
-	case "sin", "cos", "tan":
+	case "sin", "cos", "tan", "floor", "ceil", "round":
```

Comment:
You're going to need to modify this part to return the correct type. So something like this:

```
switch name {
case "sin", "cos", "tan":
    return influxql.Float, nil
case "floor", "ceil", "round":
    switch args[0] {
    case influxql.Float, influxql.Integer, influxql.Unsigned, influxql.Unknown:
        return args[0], nil
    default:
        return nil, fmt.Errorf("invalid argument type for first argument in %s(): %s", name, args[0])
}
```

The same or a similar error should be returned for the trig functions, but I'll take care of that so we can ignore that for now. The important part is you need to whitelist OK types. Unknown should be considered an OK type since it _might_ be correct, but just not present in the database.